### PR TITLE
fix document, example readme and code

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

bug fix

## Description

Readme is not correct to fit new design and enhance example capability to support more models.

## Expected Behavior & Potential Risk

Existing test pass.

## How has this PR been tested?

local test on other kind of model in torchvision, such as densenet121.

### CODE REVIEW

**Review by violetch24 — APPROVED** [review]

File: docs/source/smooth_quant.md [hunk 1]
```diff
@@ -351,9 +351,9 @@ smooth_quant_args description:
 
 "alpha": "auto" or a float value. Default is 0.5. "auto" means automatic tuning.
 
-"folding":
-- False: Allow inserting mul to update the input distribution and not absorbing. IPEX can fuse inserted mul automatically and folding=False is recommended. And for PyTorch FBGEMM backend, folding=False setting will only convert model to QDQ model.
-- True: Only allow inserting mul with the input scale that can be absorbed into the last layer. 
+"folding": whether to fold torch.mul into the last layer.
```
Comment (hshen14) [inline docs/source/smooth_quant.md:354]: it's not only for pytorch. we may remove torch

File: docs/source/smooth_quant.md [hunk 2]
```diff
@@ -351,9 +351,9 @@ smooth_quant_args description:
...
+- False: Allow inserting mul to update the input distribution and no folding. IPEX (version>=2.1) can fuse inserted mul automatically and folding=False is recommended. For Stock PyTorch, setting folding=False will only convert the model to a QDQ model.
```
Comment (hshen14) [inline docs/source/smooth_quant.md:355]: How about below?
False: Stock PyTorch and Intel Extension for PyTorch (version < 2.1)
True: Intel Extension for PyTorch (version >= 2.1) and ONNX Runtime?

File: docs/source/smooth_quant.md (hunk 1 above)
Comment (xin3he) [reply docs/source/smooth_quant.md:354]: Yes, I agree.

File: docs/source/smooth_quant.md (hunk 2 above)
Comment (hshen14) [reply docs/source/smooth_quant.md]: 1. It's framework backend capability. so the better way is to define such config in backend yaml. User should not care about folding at front-end unless we want to override backend capability. Shall we do that?
2. If we agree with 1, we don't need to worry about 2 in user config. but we may need to give a support matrix for SmoothQuant across frameworks and backends in SmoothQuant readme. Please talk with Mengni.

Comment (xin3he) [reply docs/source/smooth_quant.md]: Okay

**Review by hshen14 — APPROVED** [review]
LGTM

**Pull request merged**